frontend/src/routes.js

```diff
@@ -19,29 +17,31 @@ const api_url = process.env.API_URL;
 
 // Add routes
 app.get('/', async (req, res) => {
-    const response = await fetch(api_url+'api/',{method:'GET',headers:{}})
+    const response = await fetch(api_url+'/api/',{method:'GET',headers:{}})
     const data = await response.json();
     res.render('homePage', {test: data}); 
 
   });
 
 app.get('/job-roles', async (req, res) => {
-    const response = await fetch(api_url+'job-roles',{method:'GET',headers:{}})
+    const response = await fetch(api_url+'/job-roles',{method:'GET',headers:{}})
     const roleData = await response.json();
     res.render('jobRolesPage', {items: roleData}); 
-
 });
 
 app.get('/job-roles/:id', async (req, res) => {
     var id = req.params.id;
-    console.log(api_url+'job-roles/'+id)
-    const response = await fetch(api_url+'job-roles/'+id,{method:'GET',headers:{}})
+    console.log(api_url+'/job-roles/'+id)
+    const response = await fetch(api_url+'/job-roles/'+id,{method:'GET',headers:{}})
     const roleData = await response.json();
-    const bandResponse = await fetch(api_url+'bands/'+roleData.bandID,{method:'GET',headers:{}})
+    const specResponse = await fetch(api_url+'/job-roles/job-spec/'+id,{method:'GET',headers:{}})
+    const specData = await specResponse.json();
+    const bandResponse = await fetch(api_url+'/bands/'+roleData.bandID,{method:'GET',headers:{}})
     const bandData = await bandResponse.json();
-    console.log({role: roleData, band: bandData})
-    res.render('jobRolePage', {role: roleData, band: bandData}); 
+    console.log({role: roleData, band: bandData, specification: specData})
+    res.render('jobRolePage', {role: roleData, band: bandData, specification: specData}); 
```

Comment:
don't need specification as a separate list

Reply:
You're now hitting the back end three times in order to serve data for one front end request. Would it be pragmatic to update the backend to return everything you need for this page in one response?